src/groups/group_action.jl

```diff
@@ -115,14 +115,15 @@ Compute the value of differential of action [`AbstractGroupAction`](@ref) `A` on
 where element `a` is acting on `p`, with respect to the group element.
 
 Let ``\mathcal G`` be the group acting on manifold ``\mathcal M`` by the action `A`.
-The action is of element ``g ∈ \mathcal G`` on a point ``p ∈ \mathcal M``.
-The differential transforms vector `X` from the tangent space at `a ∈ \mathcal G`, 
-``X ∈ T_a \mathcal G`` into a tangent space of the manifold ``\mathcal M``.
-When action on element `p` is written as ``\mathrm{d}τ^p``, with the specified left or right
-convention, the differential transforms vectors
+The action is of element ``a ∈ \mathcal G`` on a point ``p ∈ \mathcal M``,
+and is denoted ``τ^p(a)``.
+The tangent map ``T τ^p`` thus transforms a vector ``X ∈ T_a \mathcal G``
```

Comment:
$Tf$ for the tangent map of $f$ is a very usual notation. It is also a "functor", as $Tf$ is the lift of $f\colon M \to N$ to $Tf \colon TM \to TN$. So $T$ acts both on functions ($Tf$) and manifolds ($TM$) (because it is a functor).

 The differential ``d`` has another meaning and is not defined when the codomain is a manifold, as far as I know.

For "transforms", I kept the original text. "map" is perhaps more adequate?

Reply:
Oh I did not think of the differential, to me that reads more like a push forward?

Concerning T itself – if that is added to the notation it is fine with me to keep that notation, sure, but we should avoid having several notations “for the same thing” – that is also why we started the notations page – to keep a bit track of that (for us even).

Reply:
Whether we write transforms, or maps – I still have no clue what that refers to :/

Reply:
("Jacobian" is also used a lot amongst engineers, but it usually refers to the matrix representation of the tangent map when basis are chosen, a useful distinction for users of `Manifolds.jl`, since `apply_diff_group` really computes the tangent map, not the matrix.)

Reply:
Yes, "push-forward" is also a possibility (since it is a covariant functor). One could use both, if you think it's clearer?

(I mention differential because it is the notation $d$ that was use prior to my change. The notation $d$ usually refers to the differential, which is not defined in the particular case.)

Reply:
Hm, you lost me again. If I have a map like ``\tau_p``, why should the differential not be defined here? I think it was not me writing that here but I am sure the author had a good idea why to write it like that.

I currently mainly try to avoid introducing another notation for things that we have (and stay unified in the notation that we do use) – and not introduce more confusion in the docs

Reply:
The differential of $f \colon \mathcal{M} \to \mathcal{G}$ at $x\in\mathcal{M}$ is a linear map $\mathrm{d}f \colon T_{x}\mathcal{M} \to \mathfrak{g}$. It is distinct (but related to) the *tangent map* (or "Jacobian", or maybe "derivative") of $f$.

Again, when the codomain of $f$ is a vector space (which is a commutative group), they can be conflated together, otherwise, not.

If you don't like the notation $T$, you could also use $D$, if you prefer, but I think $Tf$ is the right one, in this situation.

Reply:
If groups are involved, then `apply_diff_group` most definitely does *not* compute the differential, just the tangent map (or Jacobian).

Reply:
Could you explain in a bit more detail what is the difference between this T functor and pushforward? I've tried using terminology from https://en.wikipedia.org/wiki/Pushforward_(differential) and did not consider how the group structure could influence the meaning of differentiation.

Reply:
I do not have a strong opinion on the notation ``T``. I just slowly feel I should not be a developer of a package of Lie groups since I do not understand what you are writing – in the sense of – at all.
I have no clue what you mean by “conflate” and I would like to use new notation only carefully – since we already have a lot a lot of different notations.

Reply:
I would maybe propose the following – we give you access to the Julia Manifolds group. You start a `LIeGroups.jl` package that “equips” manifolds from `Manifolds.jl` with a Lie group operation / action and all necessary stuff.
That way we split both parts code wise (which is usually a good thing in Julia), we do not have so many discussions and can – when that package is mature enough – remove the Lie groups here and have a separate  package that does that better by then.

Because sure – I could take a sabbatical and read all about Lie groups before we continue here – but I do not have that time any time soon.

Reply:
Oh, I see, then the wikipedia page contradicts what I just explained. 😆 But it will probably change (look at the discussion, under "Should we abandon the d \mathrm{d} notation in favour of T \mathrm {T} or ′ '?").

A pushforward, in general, for me, is related to some covariant functor (which $T$ is). But you could say that a function "pushes forward" a probability distribution, for instance, it's another push-forward (and another covariant functor). A "lift" is another term. "Push-forward" is usually used in combination with their evil twin, "pull-backs", associated to *contravariant* functors.

A standard notation is sub-star for push-forwards, so $Tf \equiv f_*$. Used in many differential geometry books.

A standard reference for any of this is "Abraham, Marsden, Ratiu - Manifolds, Tensor Analysis and Applications".

Reply:
Another reason not to confuse them, even for a real valued function $f\colon\mathcal{M}\to\mathbb{R}$: $\mathrm{d}(\mathrm{d}f) = 0$, but $T (T f) \neq 0$ in general.

Reply:
If you find $T$ too "category theoretical", you could go for $D$, just not $d$, because of the confusion with the differential (despite the wikpedia page 😄).

Reply:
Well, i see there is a huge confusion around those terms on the internet. I suggest you choose your notations/terms and stick with it, I'm not going to interfere. 😄

[This page](https://joachimweise.github.io/post/2020-05-01-derivative/) is interesting, for instance, if you want a reference about differential, push-forward, etc.

Reply:
Yes that you complete explanation contradicts Wikipedia is one of the things I struggle with – I do not think it will change, since the point you refer to where they discussed that is from 2017.

I am fine with a $D$ for the differential – cf. Sec. 8.5 of https://www.nicolasboumal.net/book/IntroOptimManifolds_Boumal_2023.pdf – is fine with me.
I still do not understand the difference then to $T$, but the only problem I see in that is that then I am maybe not the audience these doc strings are meant for?

Reply:
Hm your page introduces push-forward or derivative as to what my link calls differential – all denoted e.g. by D and while you are fine with push-forward you are not fine with differential? That is where I am lost. Both your link and my link define the same thing the same way.

Reply:
> I do not have a strong opinion on the notation `T`. I just slowly feel I should not be a developer of a package of Lie groups since I do not understand what you are writing – in the sense of – at all. I have no clue what you mean by “conflate” and I would like to use new notation only carefully – since we already have a lot a lot of different notations.

Ok, I'm sorry I was not clear enough. 😢

It's just that there are different mathematical objects that shouldn't be confused.

If the manifold happens to be $N = \mathbb{R}$, and you have $f \colon M \to N$, then the *tangent map* is a linear map from $T_xM$ to $T_{f(x)}N$, so the tangent space at $f(x)$ on $\mathbb{R}$. Of course, you can identify this tangent space to $\mathbb{R}$ itself (more precisely: the tangent space of $\mathbb{R}$ at zero) and you then obtain the *differential* of $f$, but the two objects, $Tf$ and $df$ are, strictly speaking, not the same. Is that clearer?

On a Lie group, there are *two* ways to identify a vector to the tangent space at identity, so there are two differential. It's used all over the place by people who use `Manifolds.jl`, in robotics, for instance, so I think it's kind of important.

Reply:
> Because sure – I could take a sabbatical and read all about Lie groups before we continue here – but I do not have that time any time soon.

You wouldn't need to know all about Lie groups (I don't). Just that it's a manifold where some operations are defined, that's it.

Why not split `Manifolds.jl` with a separate `LieGroups.jl`, it could make sense, I'm not sure.

Reply:
> Yes that you complete explanation contradicts Wikipedia is one of the things I struggle with – I do not think it will change, since the point you refer to where they discussed that is from 2017.
> 
> I am fine with a D for the differential – cf. Sec. 8.5 of https://www.nicolasboumal.net/book/IntroOptimManifolds_Boumal_2023.pdf – is fine with me. I still do not understand the difference then to T, but the only problem I see in that is that then I am maybe not the audience these doc strings are meant for?

Use whatever notation you think is easiest on the users. There is a huge confusion about all this anyway.

I am sorry to have changed the notations. Feel free to ignore the pull-request if it adds to much confusion.

Reply:
I think this is an important discussion, and one of my personal goals in JuliaManifolds is clearing some confusion how different differential-geometric concepts correspond to concrete numerical calculation, and having a consistent description of differentiation is important for that goal.

By the way, since we are already discussing this, do you know any good sources that discuss higher differentials/derivatives/tangent maps/push-whatevers? Some people formalize them using higher tangent bundles, while others use jet bundles which are quite hard to understand to me. So, when clearing this topic, we could do it with such concepts in mind I think.

Reply:
> Why not split `Manifolds.jl` with a separate `LieGroups.jl`, it could make sense, I'm not sure.

The main reason not to do that currently is, that we are not enough people to get that started.

Reply:
> I think this is an important discussion, and one of my personal goals in JuliaManifolds is clearing some confusion how different differential-geometric concepts correspond to concrete numerical calculation, and having a consistent description of differentiation is important for that goal.

I agree to this. A consistent notation and documentation that helps users to use `Manifolds.jl` should be the main goal here.

Reply:
> By the way, since we are already discussing this, do you know any good sources that discuss higher differentials/derivatives/tangent maps/push-whatevers? Some people formalize them using higher tangent bundles, while others use jet bundles which are quite hard to understand to me. So, when clearing this topic, we could do it with such concepts in mind I think.

No, not really... Both higher tangent bundles and jet bundles give me headaches. 😃

A reference on jet bundles is the impenetrable (to me) [book by Peter Michor](https://www.mat.univie.ac.at/~michor/kmsbookh.pdf).

I think it is more helpful to start from a particular *application* involving higher order derivatives (perhaps related to something that can be computed with `Manifolds.jl`) and take it from there.

For instance, it is fairly common to need the Hessian in optimisation, but then you need some Riemannian structure, and all becomes clearer.

(Digression: why are higher tangent bundle so complicated?

If you lift a curve once, it can "fill" the first tangent bundle. That's easy. That is, a point of $TM$ can be made a tangent to a curve.

Now, if you lift a curve *twice* (take the second order derivative), it will not fill up the second order bundle. In other words: the second order bundle has hidden canonical structure. It just gets worse with higher order bundles, I guess.

What are the implications for computations? I don't know.)

Reply:
Regarding application, I was considering what kind of higher order differentiation would be the right formalism for second order rigid body dynamics (i.e. rigid body under constant linear and angular acceleration). So far nothing I've somewhat understood feels right, and I'm not quite happy that string theory appears on the first page of Google search for most related queries I make :disappointed: .

Reply:
Yes, that was one option I was considering; what I don't like about it is that the iterated tangent bundle has too many degrees of freedom. A force in $T^{\*}_zP$ is described by a point $p\in M$, a cotangent vector $\xi \in T^\*_p M$, and cotangents corresponding to $p$ and $\xi$. $\xi$ and the cotangent corresponding to $p$ are not independent in our problem but this formalism separates them.

I will take a look at the book you linked, maybe there is something useful there.

Reply:
Back to the notation. I think we should (a) carefully introduce non-standard notation and if so work thoroughly with references (we have `DocumenterCitations`now) – and _if_ we introduce / adapt some notation, we should do that in _all_ documentation, not just in a single doc string.